app/models/settings/base.rb

```diff
@@ -153,7 +153,7 @@ def all_settings
 
     # get the setting's value, YAML decoded
     def value
-      YAML.load(self[:value]) if self[:value].present? # rubocop:disable Security/YAMLLoad
```

Comment:
probably safe but this sneaked in here?

Reply:
This came from @heliocola's original commit; I don't know too deeply whether or not it should be here, or what the original rationale was. Maybe HC can chime in here?

Reply:
It's probably a lint fix that came with the ruby upgrade.

Reply:
While 3.1.4 is relatively mild ([updated URI and Time to patch vulnerabilities](https://www.ruby-lang.org/en/news/2023/03/30/ruby-3-1-4-released/)), we're leaping all the way from `3.0.6` to `3.1.4` here, and `YAML` changes are probably the biggest change in Ruby 3.1. There's a nice write-up over at https://www.ctrl.blog/entry/ruby-psych4.html, but `tl/dr`:

> For over a decade, Psych’s load() and load_file() method have (in practice) been aliases for the unsafe_load() and unsafe_load_file() methods. In version 4, these methods have changed to their safe_ prefixed equivalents by default.

> A safe load is better than an unsafe one, right? Yeah, it’s probably a good call to change the default. However, it has also broken a large number of Ruby packages.

Or, `tldr-ier`: pre-4 `load` _is_ `unsafe_load`, 4 changed `load` to `safe_load`. This particular invocation could _probably_ change to `safe_load` — from what I understand of how we're using it, I mean — but I _think_ we can attempt that change in a separate PR, just to minimize the number of variables in play.

Reply:
Thanks a bunch for all the link-outs here for explanations. I agree with the sentiment to do unsafe -> safe as a (potentially-fast-) follow PR separate from this to reduce risk.

Reply:
The reason I've changed this to `unsafe_load`, needed for Ruby 3.1.4, is because the Rails app is configured to use `unsafe_load`.
Ref: https://github.com/forem/forem/blob/c0b65c248c52815e4437a2bbdfdb54c8eb3a8c3a/config/application.rb#L59

I agree with @klardotsh that this could go in its own separate PR.